src/app/src/main/kotlin/org/vaccineimpact/api/app/controllers/UserController.kt

```diff
@@ -14,30 +14,29 @@ class UserController(context: ControllerContext) : AbstractController(context)
             SecuredEndpoint("/", this::getUsers, setOf("*/users.read"))
     )
 
-    fun getUser(context: ActionContext): UserInterface
+    fun getUser(context: ActionContext): User
     {
         val userName = userName(context)
-
-        val roleReadingPermissions = context.permissions.filter { p -> p.name == "roles.read" }
-
-        if (!roleReadingPermissions.any())
-            return repos.user().use { it.getUserByUsername(userName) }
-
-        val userWithRoles = repos.user().use { it.getUserWithRolesByUsername(userName) }
-
-        userWithRoles.roles = userWithRoles.roles.filter { r ->
-            roleReadingPermissions.any {
-                p ->  p.scope.encompasses(Scope.parse(r))
-            }
+        val roleReadingScopes = context.permissions
+                .filter { it.name == "roles.read" }
+                .map { it.scope }
+
+        val user = repos.user().use { it.getUserByUsername(userName) }
+        if (roleReadingScopes.any())
+        {
+            val allRoles = repos.user().use { it.getRolesForUser(userName) }
+            val roles = allRoles.filter { roleReadingScopes.encompass(Scope.parse(it)) }
+            return user.copy(roles = roles)
+        }
+        else
+        {
+            return user
         }
-
-        return userWithRoles
     }
 
```

Comment:
I also made some changes here with the goal (hopefully) of improving clarity:

* I flipped around the if statement, to make it read positive condition first
* I noticed that we only needed to know about the scopes for the `roles.read` permission, and then that led me to be able to simplify the role filter logic.